src/safe_queue.h

```diff
@@ -178,14 +174,7 @@ class safeQueue
     auto elem = queue.front();
     queue.pop();
 
-    if (queue.size() < limit)
-      is_full = false;
-
-    if (queue.size() > 0) {
-      cv.notify_one();
-    }
-
-    return elem;
+    return std::move(elem);
```

Comment:
Copy elision should handle this for you, I'm a little surprised the compiler didn't issue a warning about it (since explicitly calling move prevents elision).

Reply:
Somre reading:
https://stackoverflow.com/questions/14856344/when-should-stdmove-be-used-on-a-function-return-value
https://en.cppreference.com/w/cpp/language/copy_elision